pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldby"`
+	// How long the lock is valid.
+	LeaseTime uint64 `json:"duration"`
+	// When the lock was first acquired.
+	AcquiredTime string `json:"atime,omitempty"`
+	// When the lock was last renewed.
+	RenewTime string `json:"rtime,omitempty"`
```

Comment:
LeaseTime, AcquireTime, RenewTime are will not sufficient to guarantee to a lease holder proper serialization of resource access. To guarantee serialization the lease holder needs to acquire a sequence number (e.g. resource version in kubernetes or index in etcd) and pass it with mutating rpcs so that the lease can be checked during the commit. Exposing these as actionable datapoints will lead to missuse of the API.

Sidenote: AcquiredTime and RenewTime should be part of Status, not Spec. See: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#spec-and-status